docker-compose.yaml

```diff
@@ -1,11 +1,14 @@
 version: '3.3'
 services:
   triton:
-    image: moyix/triton_with_ft:22.09
+    build:
+      context: .
+      dockerfile: Dockerfile
     command: bash -c "CUDA_VISIBLE_DEVICES=${GPUS} mpirun -n 1 --allow-run-as-root /opt/tritonserver/bin/tritonserver --model-repository=/model"
     shm_size: '2gb'
     volumes:
       - ${MODEL_DIR}:/model
+      - ${HF_CACHE_DIR}:/root/.cache/huggingface
```

Comment:
If no `HF_CACHE_DIR` is set this breaks the deployment. Maybe we should default to true?

Reply:
The current version sets HF_CACHE_DIR to `/tmp/hf_cache`. The reason I didn't set it to default to true was because docker messes up the file permissions by setting many of them to root (unless rootless docker is used). So, the cache is shared only if the user knows what they're doing.

I could default it to true, and warn the user about the permission issues. Not sure which is a good option.

@fdegier, @moyix thoughts?

Reply:
@thakkarparth007 if a user does not use the cache, the volume is set to empty, which can not be mounted and causes the `docker compose up` to fail, hence why I suggested to default to true but I understand the permission issues you mentioned. 

I think adding  something like `HF_DATASETS_CACHE="fauxpilot/.hf-cache"` and removing the option to cache, will always cache and not mess up permissions?

Reply:
Whoops, I just saw your comment @fdegier 

So currently if you notice in the setup.sh (https://github.com/moyix/fauxpilot/pull/86/files#diff-4209d788ad32c40cbda3c66b3de47eefb929308ca703bb77a6382625986add17R148) then you'll see that HF_CACHE_DIR is being set to `/tmp/hf_cache` if the user doesn't want to share hf_cache.

But yes, perhaps it'll be better to store the cache in the fauxpilot directory itself.